core/core/services/contact.py

```diff
@@ -21,3 +23,36 @@ def add_contact(payload: ContactPayload) -> Record:
         note=payload.note,
         birthday=birthday,
     )
+
+
+@response(InfoMessages.CONTACT_UPDATED)
+@write_data
+def update_contact(payload: ContactPayload) -> Record:
+    record = database[payload.name]
+
+    if record:
+        if payload.name:
+            record.name = payload.name
+        if payload.email:
+            record.email = payload.email
+        if payload.note:
+            record.note = payload.note
+        if payload.birthday:
+            record.birthday = datetime.strptime(payload.birthday, "%d.%m.%Y").date()
+        if payload.address:
+            record.address = payload.address
+        if payload.phones:
+            record.phones = payload.phones
+        if payload.tags:
+            record.tags = payload.tags
+        return record
+
+
+@response(InfoMessages.CONTACT_FOUND)
+def check_contact(payload):
+    record = database[payload.name]
+
+    if record:
+        return record
+    else:
+        raise KeyError
```

Comment:
I think we can just return `None` if there is no contact in db, usually it works that way in real databases